controllers/imagerepository_notifications.go

```diff
@@ -0,0 +1,198 @@
+						if resStatusNotification.UUID == statusNotificationAux.UUID {
+							existsInStatus = true
+							break
+						}
+					}
+					if !existsInStatus {
+						imageRepository.Status.Notifications = append(imageRepository.Status.Notifications, resStatusNotification)
+					}
+				}
+			}
+		}
+		if !existsInSpec {
+			// Deleted item from Spec.Notifications: delete notification from Quay and Status.Notifications
+			log.Info("Deleting notification in Quay", "Title", statusNotification.Title, "UUID", statusNotification.UUID)
+			_, err := r.QuayClient.DeleteNotification(
+				r.QuayOrganization,
+				imageRepository.Spec.Image.Name,
+				statusNotification.UUID)
+			if err != nil {
+				log.Error(err, "failed to delete notification", "Title", statusNotification.Title, "UUID", statusNotification.UUID)
+				return r.handleError(ctx, imageRepository, err, "Error while deleting a notification ("+statusNotification.Title+") to Quay")
+			}
+			// Remove notification from CR status
+			imageRepository.Status.Notifications = append(imageRepository.Status.Notifications[:index], imageRepository.Status.Notifications[index+1:]...)
+			index--
+		}
+	}
+
+	return nil
+}
+
+// This function adds all Spec.Notifications to Quay and overwrites all existing notifications in ImageRepository Status
+func (r *ImageRepositoryReconciler) SetNotifications(ctx context.Context, imageRepository *imagerepositoryv1alpha1.ImageRepository) ([]imagerepositoryv1alpha1.NotificationStatus, error) {
+	log := ctrllog.FromContext(ctx).WithName("ConfigureNotifications")
+
+	if imageRepository.Spec.Notifications == nil {
+		// No notifications to configure
+		return nil, nil
+	}
+
+	log.Info("Adding notifications")
+	notificationStatus := []imagerepositoryv1alpha1.NotificationStatus{}
+
+	for _, notification := range imageRepository.Spec.Notifications {
+		log.Info("Creating notification in Quay", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+		notificationStatusRes, err := r.AddNotification(notification, imageRepository)
+		if err != nil {
+			log.Error(err, "failed to create notification", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+			resErr := r.handleError(ctx, imageRepository, err, "Error while adding a notification ("+notification.Title+") to Quay")
+			return nil, resErr
+		}
+		notificationStatus = append(notificationStatus, notificationStatusRes)
+		log.Info("Notification added",
+			"Title", notification.Title,
+			"Event", notification.Event,
+			"Method", notification.Method,
+			"QuayNotification", notificationStatusRes)
+	}
+	return notificationStatus, nil
+}
+
+func (r *ImageRepositoryReconciler) notificationExistsInQuayByUUID(UUID string, imageRepository *imagerepositoryv1alpha1.ImageRepository) (quay.Notification, error) {
+	notification := quay.Notification{}
+	allNotifications, err := r.QuayClient.GetNotifications(r.QuayOrganization, imageRepository.Spec.Image.Name)
```

Comment:
This is called at each iteration, why have you reverted fetching all notifications before the loop?

Reply:
Explained via Slack right now

Reply:
Discussed in a meeting, count removed.

Reply:
Discussed in a meeting, the approach has to be optimized.